Split gpl package

hmaarrfk opened the pull request:
Superseeds #30 (suggested using libedit instead)
Fixes #29 

Splits the package into a GPL'ed portion (the binary) and the non-gpl'ed portion (the library). See results from [ldd](https://github.com/conda-forge/sqlite-feedstock/pull/30#issuecomment-414130501).

(I didn't test it with windows or mac, lets see what the CI's say)

The binary is compiled with readline support, so it must have a GPLv3 license.

Notes from the discussion in #30
@bgruening suggested looking at how [debian packages things](https://github.com/conda-forge/sqlite-feedstock/pull/30#issuecomment-414131251)

I am unfamiliar with debian packaging, but it doesn't seem that they list a license in their package.

They they split their package into an executable package and library.


Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy`
* [ ] Ensured the license file is being packaged.


XREF:
Issue on Continuum repo: 
* https://github.com/ContinuumIO/anaconda-issues/issues/307
* https://github.com/ContinuumIO/anaconda-recipes/issues/27

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found some lint.

Here's what I've got...


For **recipe**:

 * The home item is expected in the about section.
 * The license item is expected in the about section.
 * The summary item is expected in the about section.
 * Recipe maintainers should be a json list.
 * Selectors are suggested to take a ``<two spaces>#<one space>[<expression>]`` form. See lines [66]
 * The outputs section contained an unexpected subsection name. tests is not a valid subsection name.

For **recipe**:

 * It looks like the 'libsqlite' output doesn't have any tests.

hmaarrfk commented:
@conda-forge-admin please lint

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

hmaarrfk commented on recipe/meta.yaml:
```diff
@@ -25,23 +25,52 @@ requirements:
+      - include/sqlite3*          # [not win]
+      - share/man/man1/sqlite3*   # [not win]
+      - lib/libsqlite3*           # [not win]
+      - lib/pkgconfig/sqlite3.pc  # [not win]
```
@jakirkham I feel like all these lines are just going to cause future pain for packagers. Is there a way to simplify things?

hmaarrfk commented on recipe/meta.yaml:
```diff
@@ -25,23 +25,52 @@ requirements:
+  - name: sqlite
+    files:
+      - bin/sqlite3               # [not win]
+      - Library/bin/sqlite3.exe   # [win]
```
@jakirkham I really just want to take the "compliment" of these files.

hmaarrfk replied:
I guess now it is pretty clean.

hmaarrfk commented:
@jakirkham @ocefpaf @scopatz @bgruening

Thank you for your comments on PR #30. I think this PR addresses all the issues and takes a step in the right direction in terms of compliance.

Depending on the feedback I receive from you guys I'll add a GPL license file for the `sqlite3` executable.

scopatz commented on recipe/meta.yaml:
```diff
@@ -25,23 +25,55 @@ requirements:
+        - sqlite3 --version
+    about:
+      home: http://www.sqlite.org/
+      license: GPL-3.0  # [not win]
```
I don't think that it is OK to change the license here.

hmaarrfk replied:
You mean the win/not win flag?

scopatz replied:
I mean that I don't think it is OK to advertise that a license covers a package that the original authors did not intend

scopatz replied:
That is, readline is clearly GPL'd and so SQLite should retain its license

scopatz commented on recipe/meta.yaml:
```diff
@@ -25,23 +25,55 @@ requirements:
+# By compiling with readline, the /bin/sqlite3 executable becomes
+# subject to the GPL
+# Therefore, we decide to split it to make it clear
+# That the other software is not.
```
I don't think that this is strictly true, or at least it is open to legal interpretation. The GPL certainly claims that (dynamically) linking will make the joint program GPL.  However, this does not necessarily follow, as when you link the only part of the code that is used is the API, which is non-copyrightable.

hmaarrfk replied:
That is fair, but there is no compatible library that has a compatible API (yet) on conda-forge so that argument is rather weak.

I think that along with the dependency on `readline`, it is hard to argue against the fact that the final product, that arrives on the user's computer is not using GPL code.

hmaarrfk commented:
@scopatz, hopefully I incorporated all the mods you requested.

The script is also much improved to ensure that we don't need to keep updating things. Let me know what you think.

hmaarrfk commented:
Superseeded by https://github.com/conda-forge/staged-recipes/pull/6550
